### PR TITLE
draft version

### PR DESCRIPTION
**Pull request by brechtbilliet**

### CODE REVIEW

File: _posts/2018-04-13-do-we-really-need-redux.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+---
+layout: post
+title: Do we really need Redux
```
Comment (KwintenP) [inline _posts/2018-04-13-do-we-really-need-redux.md]: or NgRX/Store

maybe add this.

File: _posts/2018-04-13-do-we-really-need-redux.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+subclass: 'post'
+categories: 'brechtbilliet'
+disqus: true
+tags: Redux
+cover: 'assets/images/cover/cover1.jpg'
+---
+
+## About this article
+
+If you are writing Angular, Vue or React applications. Chances are big that you have used or encountered the Redux pattern.
```
Comment (KwintenP) [inline _posts/2018-04-13-do-we-really-need-redux.md]: applications, chances

Seems that this should be a comma.

File: _posts/2018-04-13-do-we-really-need-redux.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+In this article I would like to tackle the question on when to use Redux and why.
+The referred plugins/code samples are written in an Angular context. But the principles explained in this articles work for other frameworks as well. This also means that we are using the [@ngrx/store](https://github.com/ngrx/platform/blob/master/docs/store/README.md) package instead of the [Redux](https://redux.js.org/) package
+
+## To Redux or not to Redux?
+
+First of all it's important to note that Redux solves certain problems for us. If we are not experiencing those problems, then there is a big chance that the Redux pattern
+is overkill for our application.
+
+The first question that we might want to ask ourselves is:
+**Does My application have state?** State can be the value of a pager that we want to remember, or the fact that a sidebar is collapsed or not. State could be a cached set of data coming from our backend, or userinformation that we need throughout the whole application.
```
Comment (KwintenP) [inline _posts/2018-04-13-do-we-really-need-redux.md]: user information

File: _posts/2018-04-13-do-we-really-need-redux.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+It could be a simple value that we want to remember in memory when we are navigating between pages.
+
+Let's sum up some examples where the Redux principle might shine in our applications:
+- Storing state (like the value of a search filter so it’s still available when the user navigates back to certain grid)
```
Comment (KwintenP) [inline _posts/2018-04-13-do-we-really-need-redux.md]: to a certain

File: _posts/2018-04-13-do-we-really-need-redux.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+- Sharing state between components that have their own route, and thus won’t have a parent component to pass them the state through inputs or properties
+- Optimistic updates: Check [this article](https://blog.strongbrew.io/Cancellable-optimistic-updates-in-Angular2-and-Redux/)
+- Real-time updates: Check [this article](https://blog.strongbrew.io/How-we-made-our-app-real-time-in-6-lines-of-code/)
+- When we want undo/redo logic
+- When we want to keep track of all the state changes and debug them with awesome tooling (Redux devtools)
+- When we want an organised way of handling session storage or localstorage. Check [this plugin](https://github.com/btroncone/ngrx-store-localstorage)
```
Comment (KwintenP) [inline _posts/2018-04-13-do-we-really-need-redux.md:39]: Couldn't you just wrap this yourself in a service? I wouldn't recommend using Redux if the only thing you need it for is localstorage. It's a nice to have imo.

File: _posts/2018-04-13-do-we-really-need-redux.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+    setUsers(users: User[]): void {
+        this._users$.next(...users);
+    }
+
+    addUser(user: User): void {
+        // We cannot use array.push because we only want to
+        // pass immutable data to the streame
+        // for the OnPush strategy remmber?
+        this._users$.next([...this._users$.getValue()])
```
Comment (KwintenP) [inline _posts/2018-04-13-do-we-really-need-redux.md]: Shouldn't this be:

`this._users$.next([...this._users$.getValue(), user])`

File: _posts/2018-04-13-do-we-really-need-redux.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
+---
```
Comment (KwintenP) [inline _posts/2018-04-13-do-we-really-need-redux.md:1]: One extra example you could add is for caching. People put a lot of stuff in the store cause they want to cache it while you can simple use a `shareReplay` operator on a stream, add it the stateService you describe below and that works as well.

**Review by KwintenP — CHANGES_REQUESTED** [review]

File: _posts/2018-04-13-do-we-really-need-redux.md (hunk 5 above)
Comment (brechtbilliet) [reply _posts/2018-04-13-do-we-really-need-redux.md:39]: that why I say: "an organised way" If you need the actions to see what happens etc

**Pull request merged**